AutorestSwiftTest/AutoRestResourceFlatteningTest.swift

```diff
@@ -44,21 +44,304 @@ class AutoRestResourceFlatteningTest: XCTestCase {
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.getWrappedArray failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_putWrappedArray() throws {
+        let expectation =
+            XCTestExpectation(
+                description: "Call autorestresourceflatteningtestservice.putWrappedArray succeed"
+            )
+        let wrapped = [
+            WrappedProduct(value: "test")
+        ]
+        client.autorestresourceflatteningtestservice.put(wrappedArray: wrapped) { result, httpResponse in
+            switch result {
+            case .success:
+                expectation.fulfill()
+            case let .failure(error):
+                let details = self.errorDetails(for: error, withResponse: httpResponse)
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.putWrappedArray failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_putSimpleProduct() throws {
+        let expectation =
+            XCTestExpectation(
+                description: "Call autorestresourceflatteningtestservice.putSimpleProduct succeed"
+            )
+        let product = SimpleProduct(
+            maxProductDisplayName: nil,
+            genericValue: nil,
+            odataValue: nil,
+            productId: "Product1",
+            description: "Description"
+        )
+        client.autorestresourceflatteningtestservice.put(simpleProduct: product) { result, httpResponse in
+            switch result {
+            case .success:
+                expectation.fulfill()
+            case let .failure(error):
+                let details = self.errorDetails(for: error, withResponse: httpResponse)
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.putSimpleProduct failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_putSimpleProductWithGroupingFlattenParameterGroup() throws {
+        let expectation =
+            XCTestExpectation(
+                description: "Call autorestresourceflatteningtestservice.putSimpleProductWithGroupingFlattenParameterGroup succeed"
+            )
+        let group = FlattenParameterGroup(
+            name: "name",
+            simpleBodyProduct: SimpleProduct(
+                maxProductDisplayName: nil,
+                genericValue: nil,
+                odataValue: nil,
+                productId: "Product1",
+                description: "Description"
+            ),
+            productId: "Product1",
+            description: nil,
+            maxProductDisplayName: nil,
+            genericValue: nil,
+            odataValue: "Description"
+        )
```

Comment:
This looks like a code generation error. The parameters are in the grouped parameter but still are in the signature.

Reply:
This has been fixed in the latest commit.